soepy/simulate/simulate_auxiliary.py

```diff
@@ -97,24 +106,31 @@ def pyth_simulate(
             current_states[:, 5],
             current_states[:, 6],
             current_states[:, 7],
+            current_states[:, 8],
         ]
 
         # Extract corresponding utilities
         current_log_wage_systematic = log_wage_systematic[idx]
```

Comment:
Let's move all the `reshape` stuff up here. It really impeds readability later on.

Reply:
If reshaped is moved here to avoid reshaping in lines 120-140, reshape is needed for lines 175ff. We might need to rethink the entire interface of the simulation. Since this is already referenced in an issue, I'd prefer to address it in the next PR, where we work on speed and code quality.